docs/explainer.md

```diff
@@ -9,6 +9,96 @@ See [Discourse](https://discourse.wicg.io/t/filling-the-remaining-gap-between-we
 User agents will need to carefully consider when to make the Raw Sockets API available to web applications,
 and what UI will be shown to the user.
 
+
+
+### Threat
+
+MITM attackers may inject sockets API calls into a web page.
+
+### Mitigation
+
+The API will only available in secure contexts (HTTPS).
+
+
+
+### Threat
+
+A web app might initiate connections without the user realizing.
+
+### Mitigation
+
+When initiating a connection, the user agent would show some UI.
+
+The user can be asked to specify a hostname or IP address, with an option to permit future connections to this host.
+
+
+
+### Threat
+
+Attackers may use the API to DDOS third parties.
+
+### Mitigation
+
+Connection attempts would be rate limited.
+
+Connections will only be able to be initiated after transient activation (user interaction with the page).
```

Comment:
This is good, but these can be fairly weak, as scroll events counts as transient activation I believe.

Reply:
Ack. We are following the general principle of no dialogs without transient activation.